### PR TITLE
Large changes on routing.md

### PR DESCRIPTION
**Pull request by gortiz**
As part of the documentation day, I rewrote most of the routing document in order to make it easier to understand the difference between different routing optimizations

### CODE REVIEW

**Comment by gortiz** [comment]
This is a lot of text and I would like to do a second read before merging it

File: operators/operating-pinot/tuning/routing.md [hunk 1]
```diff
@@ -1,70 +1,90 @@
...
+Apart from the ascending time, Apache Pinot can also take advantage of other distribution data patterns: Partition by segments
+In Pinot, data is partitioned by a dimension column when all rows on the same segment have the same partition value for the partitioning column.
+More formally, given a function `p`, for all segments `s`, given any pair of rows `r1` and `r2` it is true that `p(r1) = p(r2)`.
+For example, in a table configured to have 3 partitions by `memberId` column, using `modulo` as the partition function, a segment that contains a row with `memberId` = 101 may also contain another row with `memberId` = 2 and another with `memberId` = 335, but it cannot contain a row with `membermId` = 336 or `memberId` = 334.
```
Comment (snleee) [inline operators/operating-pinot/tuning/routing.md]: Found a typo :) `membermId` -> `memberId`

File: operators/operating-pinot/tuning/routing.md [hunk 2]
```diff
@@ -97,4 +115,57 @@ column.memberId.numPartitions = 4
 column.memberId.partitionValues = 1
 ```
 
-Broker side pruning for partitioning can be configured by setting the `segmentPrunerTypes` in the `RoutingConfig`. Note that the current implementation for partitioning only works for **EQUALITY** and **IN** filter (e.g. `memberId = xx`, `memberId IN (x, y, z)`).
+It is important to note that Pinot uses the partition configuration as metainformation.
+On offline tables, Pinot assumes that segments actually honor the specified partition, so the ingestion process must guarantee that.
```
Comment (snleee) [inline operators/operating-pinot/tuning/routing.md]: Can we add a bit more information on what needs to be guaranteed? When ingesting offline data, we usually expect that each input file contains all the relevant rows for a single partition.

**Review by snleee — APPROVED** [review]
Added some minor comments. Thank you for updating the document 👍

**Review by gortiz — COMMENTED** [review]
Fix typo

File: operators/operating-pinot/tuning/routing.md (hunk 2 above)
Comment (gortiz) [reply operators/operating-pinot/tuning/routing.md]: > we usually expect that each input file contains all the relevant rows for a single partition.

Is that correct? I thought what we expect is that all cells in the column that is partitioned belong to the same partition. But a single segment doesn't have to contain all rows of a single partition.

Comment (gortiz) [reply operators/operating-pinot/tuning/routing.md]: I've added an example (although it is almost a repetition from the one we have above)

**Review by Jackie-Jiang — APPROVED** [review]
This is awesome!

**Comment by Jackie-Jiang** [comment]
There are some conflicts with the base branch. Can you resolve that so that we can merge the change

**Comment by gortiz** [comment]
> There are some conflicts with the base branch. Can you resolve that so that we can merge the change

I think I solved them before you commented that. Maybe we had a race condition there ^_^. Commit [6f32ae0](https://github.com/pinot-contrib/pinot-docs/pull/134/commits/6f32ae04b74d11b99d168daf2f4ab4d661e5b218) should be mergeable without problems.

**Comment by Jackie-Jiang** [comment]
@gortiz It still doesn't allow me to merge. Can you try rebase again and force push this branch?

Screenshot 2023-03-22 at 9 59 27 AM

**Pull request merged**